worlds/sc2/client.py

```diff
@@ -1287,258 +1362,52 @@ async def updateProtossTech(self, current_items):
         protoss_items = current_items[SC2Race.PROTOSS]
         await self.chat_send("?GiveProtossTech " + " ".join(map(str, protoss_items)))
 
+def calc_unfinished_nodes(
+        ctx: SC2Context
+) -> typing.Tuple[typing.Set[int], typing.Set[int], typing.Set[int], typing.Set[int]]:
```

Comment:
The return type is incorrect (the actual, not the declared type is used)

Reply:
If I'm reading this correctly, the actual return args are a list, a dict, a list, and a set.